### PR TITLE
Add AggregateExceptionTest

### PR DESCRIPTION
**Pull request by jbagga**
Add functional test for PR: aspnet/Common#215

Checks that `AggregateException`'s inner exceptions are displayed in a `DeveloperExceptionPage`

cc @pranavkm

### CODE REVIEW

File: test/WebSites/ErrorPageMiddlewareWebSite/AggregateExceptionController.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        {
+            var firstException = ThrowNullReferenceException();
+            var secondException = ThrowIndexOutOfRangeException();
+            Task.WaitAll(firstException, secondException);
+            return View();
+        }
+
+        private static async Task ThrowNullReferenceException()
+        {
+            await Task.Delay(2);
```
Comment (pranavkm) [inline test/WebSites/ErrorPageMiddlewareWebSite/AggregateExceptionController.cs]: Does it work reliably if you do `Task.Delay(0)`?

File: test/WebSites/ErrorPageMiddlewareWebSite/ErrorPageMiddlewareWebSite.csproj [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<Project Sdk="Microsoft.NET.Sdk.Web">
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
```
Comment (pranavkm) [inline test/WebSites/ErrorPageMiddlewareWebSite/ErrorPageMiddlewareWebSite.csproj]: ?

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/AggregateExceptionTest.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        public AggregateExceptionTest(MvcTestFixture<ErrorPageMiddlewareWebSite.Startup> fixture)
+        {
+            Client = fixture.Client;
+        }
+
+        public HttpClient Client { get; }
+
+        [Fact]
+        public async void AggregateExceptionFlattensInnerExceptions()
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/AggregateExceptionTest.cs]: Add this to the ErrorPageMiddlewareTest

File: test/WebSites/ErrorPageMiddlewareWebSite/ErrorPageMiddlewareWebSite.csproj (hunk 2 above)
Comment (jbagga) [inline test/WebSites/ErrorPageMiddlewareWebSite/ErrorPageMiddlewareWebSite.csproj]: VS automatically did it. Doesn't change anything

Comment (pranavkm) [inline test/WebSites/ErrorPageMiddlewareWebSite/ErrorPageMiddlewareWebSite.csproj]: Can we undo this?

**Comment by jbagga** [comment]
@pranavkm Updated

**Pull request merged**